crates/ruff_python_semantic/src/binding.rs

```diff
@@ -248,6 +263,16 @@ impl<'a> Binding<'a> {
         })
     }
 
+    /// Returns the statement range for bindings created by `[ast::StmtAnnAssign]`
+    /// or `[ast::StmtClassDef]` and `[Binding.range]` otherwise.
+    pub fn defn_range(&self, semantic: &SemanticModel) -> TextRange {
```

Comment:
What does `defn` stand for`. Is it `definition`? If so, I suggest we use the long form (it's not that much longer) because we generally try to avoid abbrevation, because it can be unclear what they mean.

What about `FunctionDef`?

Reply:
This is a very specialized function to ensure that the following lexicographical lookups are correct:

```python
x: TypeAlias = x | None   # x is not available at runtime

class X:
    foo: TypeAlias = X   # X is not available at runtime
```

For that the entire statement needs to be used for looking up whether the binding occurred lexicographically before our lookup. Otherwise we get the wrong result. The same is not true for functions, since they can refer to themselves.

I couldn't really think of a good name for this function, maybe it would be better to inline this logic into `simulate_runtime_load`, since it's only used once.